dali/pipeline/data/types.h

```diff
@@ -427,23 +429,23 @@ class DLL_PUBLIC TypeTable {
  public:
   template <typename T>
   DLL_PUBLIC static DALIDataType GetTypeId() {
-    auto &inst = instance();
-    static DALIDataType type_id = inst.RegisterType<T>(static_cast<DALIDataType>(++inst.index_));
+    static DALIDataType type_id = instance().RegisterType<T>(
+        static_cast<DALIDataType>(instance().next_id_++));
     return type_id;
   }
 
   template <typename T>
-  DLL_PUBLIC static string GetTypeName() {
+  DLL_PUBLIC static std::string_view GetTypeName() {
     return TypeNameHelper<T>::GetTypeName();
   }
 
   DLL_PUBLIC static const TypeInfo *TryGetTypeInfo(DALIDataType dtype) {
-    auto &inst = instance();
-    std::lock_guard<spinlock> guard(inst.lock_);
-    auto id_it = inst.type_info_map_.find(dtype);
-    if (id_it == inst.type_info_map_.end())
+    auto *types = instance().type_info_map_;
+    assert(types);
+    size_t idx = dtype - DALI_NO_TYPE;
+    if (idx >= types->size())
       return nullptr;
-    return &id_it->second;
+    return (*types)[idx];
```

Comment:
What happens when `RegisterType` allocated new vector of TypeInfo pointers because of `idx >= type_info_map_->size()` and we're asking for some type registered earlier? Won't we end up with nullptr here?